kikuchipy/signals/ebsd_master_pattern.py

```diff
@@ -289,3 +407,224 @@ class LazyEBSDMasterPattern(EBSDMasterPattern, LazySignal2D):
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
+
+
+def _get_direction_cosines(detector: EBSDDetector) -> Vector3d:
+    """
+    Get the direction cosines between the detector and sample as done in EMsoft
+    and [Callahan2013]_.
+
+    Parameters
+    ----------
+    detector : EBSDDetector
+        EBSDDetector object with a certain detector geometry.
+
+    Returns
+    ----------
+    r_g.unit : Vector3d
+        Vector3d object containing the direction cosines for each detector
+        pixel.
+    """
+
+    pc = detector.pc_emsoft()
+    xpc = pc[..., 0]
+    ypc = pc[..., 1]
+    L = pc[..., 2]
+
+    # Detector coordinates in microns
+    det_x = (
+        -((-xpc - (1.0 - detector.ncols) * 0.5) - np.arange(0, detector.ncols))
+        * detector.px_size
+    )
+    det_y = (
+        (ypc - (1.0 - detector.nrows) * 0.5) - np.arange(0, detector.nrows)
+    ) * detector.px_size
+
+    # Auxilliary angle to rotate between reference frames
+    theta_c = np.radians(detector.tilt)
+    sigma = np.radians(detector.sample_tilt)
+
+    alpha = (np.pi / 2) - sigma + theta_c
+    ca = np.cos(alpha)
+    sa = np.sin(alpha)
+
+    # NYI
```

Comment:
What does "NYI" stand for?

Reply:
"Not Yet Implemented", currently _omega_ (angle between normal of sample and detector) is set to 0  for every setup.

Reply:
Aha, I suggest replacing "NYI" with a "# TODO: Enable detector azimuthal angle" or something similar, so that it is clear what is desired!